app/models/budget.rb

```diff
@@ -16,13 +16,17 @@ class Budget < ApplicationRecord
     where(budget_type: budget_types)
   }
 
+  before_validation :infer_and_assign_providing_org_attrs
+
   validates_presence_of :report, unless: -> { parent_activity&.organisation&.service_owner? }
   validates_presence_of :value,
     :currency,
     :financial_year
   validates :value, numericality: {other_than: 0, less_than_or_equal_to: 99_999_999_999.00}
   validates :budget_type, inclusion: {in: BUDGET_TYPES.values}
   validate :direct_budget_type_must_match_source_fund, if: -> { DIRECT_BUDGET_TYPES.include?(budget_type) }
+  validates_presence_of :providing_organisation_id, if: -> { DIRECT_BUDGET_TYPES.include?(budget_type) }
```

Comment:
I'm wondering if it's worth [grouping these validations](https://guides.rubyonrails.org/active_record_validations.html#grouping-conditional-validations) together?

```ruby
with_options if: -> { DIRECT_BUDGET_TYPES.include?(budget_type) } do |direct_budget|
  direct_budget.validate :direct_budget_type_must_match_source_fund
  direct_budget.validate :direct_budget_providing_org_must_be_beis
  direct_budget.validates :providing_organisation_id, presence: true
end
```

Reply:
Thank you for the suggestion! Upcoming with the next push.